common/trainer/default_trainer.py

```diff
@@ -147,12 +148,19 @@ def train(self):
                     if self.lr_scheduler is not None:
                         self.lr_scheduler.step()
                     self.optimizer.zero_grad()
-                    if step % log_step == 0:
-                        logger.info(f"train epoch:[{idx}/{num_train_epochs}]\tstep:[{step}/{total_steps}]\tloss:{loss:.6f}\tppl:{math.exp(loss):.6f}\ttime:{time.time()-start:.6f}")
-                        report({"train_epoch": idx, "total_epochs": num_train_epochs, "train_step": step, "total_steps": min(max_train_step, total_steps) if max_train_step else total_steps})
+
+                    # Checks if the accelerator has performed an optimization step behind the scenes
+                    if self.accelerator.sync_gradients:
+                        self.completed_steps += 1
+
+                    if self.completed_steps % logging_steps == 0:
+                        perplexity = math.exp(loss)
```

Comment:
Would it be better to use `loss.item()` here?

Reply:
Yes, thanks. I will update.